pandas/_typing.py

```diff
@@ -115,7 +114,7 @@
 Ordered = Optional[bool]
 JSONSerializable = Optional[Union[PythonScalar, List, Dict]]
 Frequency = Union[str, "DateOffset"]
-Axes = Collection[Any]
+Axes = Union[AnyArrayLike, List, Dict, range, Sequence[str]]
```

Comment:
Why specifically Sequence[str] and not Sequence[Hashable]?

Reply:
I've removed `Dict` and `Sequence[str]` in the next commit.  `Sequence[Hashable]` will match a string, and that is not a valid value for `index` or `columns`